tests/gpu/t_ec_sum_port.nim

```diff
@@ -0,0 +1,464 @@
+# Constantine
+# Copyright (c) 2018-2019    Status Research & Development GmbH
+# Copyright (c) 2020-Present Mamy André-Ratsimbazafy
+# Licensed and distributed under either of
+#   * MIT license (license terms in the root directory or at http://opensource.org/licenses/MIT).
+#   * Apache v2 license (license terms in the root directory or at http://www.apache.org/licenses/LICENSE-2.0).
+# at your option. This file may not be copied, modified, or distributed except according to those terms.
+
+##[
+
+NOTE:
+
+This test case was used to port the `sumImpl` template implementation for CPU targets from
+`ec_shortweierstrass_jacobian.nim` to the LLVM based GPU target.
+
+It contains multiple commented out lines of proc signatures, input types and many `asy.store`
+instructions. By using `asy.store` and adjusting the return type of the LLVM based procedure,
+(i.e. the first argument in the following line from below:
+
+`asy.llvmPublicFnDef(name, "ctt." & ed.name, asy.void_t, [ed.curveTy, ed.curveTy, ed.curveTy]):`
+                                                          ^--- First argument == return type
+
+and adjusting the `cpuSumImpl` below to return the 'same' value, one can compare all intermediary
+results line by line to verify correctness. Porting the entire code first and then trying
+to pin down small bugs is a lot more bothersome.
+
+]##
+
+import
+  # Internal
+  constantine/named/algebras,
+  constantine/math/io/[io_bigints, io_fields, io_ec],
+  constantine/math/arithmetic,
+  constantine/math/elliptic/ec_shortweierstrass_jacobian,
+  constantine/platforms/abstractions,
+  constantine/platforms/llvm/llvm,
+  constantine/math_compiler/[ir, pub_fields, pub_curves_jacobian, codegen_nvidia, impl_fields_globals],
+  # Test utilities
+  helpers/prng_unsafe
+
+proc genSumImpl*(asy: Assembler_LLVM, ed: CurveDescriptor): string =
```

Comment:
For now I've deleted it.

Reply:
Yes, of sorts. It's the file of how I ported over the CPU version. I wanted to commit it so that it wouldn't be lost (in the PR branch at least!). Can be useful both for other people trying to get an idea on how to not get overwhelmed porting a more complex algorithm or simply myself in the future.